frontend/server/src/Controllers/Course.php

```diff
@@ -95,20 +95,47 @@ private static function validateCreateAssignment(
         );
 
         $r->ensureOptionalBool('unlimited_duration');
-        $startTime = $r->ensureTimestamp(
-            'start_time',
-            $courseStartTime->time,
-            is_null($courseFinishTime) ? null : $courseFinishTime->time
-        );
-        $finishTime = $r->ensureOptionalTimestamp(
-            'finish_time',
-            $courseStartTime->time,
-            is_null($courseFinishTime) ? null : $courseFinishTime->time,
-            /*$isRequired=*/(
-                !is_null($courseFinishTime) ||
-                !$r['unlimited_duration']
-            )
-        );
+        try {
+            $startTime = $r->ensureTimestamp(
+                'start_time',
+                $courseStartTime->time,
+                is_null($courseFinishTime) ? null : $courseFinishTime->time
+            );
+        } catch (\OmegaUp\Exceptions\InvalidParameterException $e) {
+            if ($e->getMessage() === 'parameterDateTooSmall') {
+                throw new \OmegaUp\Exceptions\InvalidParameterException(
+                    'courseAssignmentStartDateBeforeCourseStartDate'
+                );
+            }
+            throw $e;
+        }
```

Comment:
para evitar el uso de [excepciones como control de flujo](https://wiki.c2.com/?DontUseExceptionsForFlowControl), se puede mejor mover la validación de la hora de inicio como un `if` explícito?

```suggestion
        $startTime = $r->ensureTimestamp(
            'start_time',
            /*$startTime=*/null,
            is_null($courseFinishTime) ? null : $courseFinishTime->time
        );
        if ($startTime->time < $courseStartTime->time) {
            throw new \OmegaUp\Exceptions\InvalidParameterException(
                'courseAssignmentStartDateBeforeCourseStartDate'
            );
        }
```

así también se es más explícito aquí en el código cuál es la condición bajo la cual se arroja esto.

igual en todos los demás lugares donde se usa este patrón.

Reply:
Listo.